src/components/Request/components/Request/index.js

```diff
@@ -0,0 +1,126 @@
+import React, {Component, PropTypes} from 'react'
+import axios from 'axios'
+import {isEqual, first} from 'lodash'
+import Loading from 'components/Loading'
+
+const http = axios.create()
+
+const methods = [
+  'get',
+  'post',
+  'put',
+  'delete',
+]
+
+export default class Request extends Component {
+
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+    method: PropTypes.oneOf(methods),
+    params: PropTypes.object,
+    headers: PropTypes.object,
+    body: PropTypes.object,
+    lazy: PropTypes.bool,
+    onResponse: PropTypes.func,
+    onData: PropTypes.func,
+    onError: PropTypes.func,
+    children: PropTypes.func,
+  }
+
+  static defaultProps = {
+    method: first(methods),
+  }
+
+  state = {
+    running: !this.props.lazy,
+    response: null,
+    data: null,
+    error: null,
+  }
+
+  componentDidMount() {
+    if (!this.props.lazy) {
+      this.request()
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!this.props.lazy && !isEqual(this.props, nextProps)) {
+      this.request()
+    }
+  }
+
+  componentWillUnmount() {
+    this.willUnmount = true
+  }
+
+  request = (body = this.props.body) => {
```

Comment:
So when there is an `=` inside a function param, it's called a "default parameter"; it is the ES2015+ way of doing `var body = body || this.props.body`, so if `<Request` has a `body` prop passed, use it, if not keep it undefined. Make sense? Here is an egghead.io lesson on it from John: https://egghead.io/lessons/ecmascript-6-default-values-for-function-parameters